src/cromshell/logs/command.py

```diff
@@ -57,222 +76,281 @@ def main(
         else str(status).strip(",").split(",")
     )
 
-    command_setup_utils.resolve_workflow_id_and_server(
-        workflow_id=workflow_id, cromshell_config=config
-    )
-
     LOGGER.info("Status keys set to %s", status_param)
 
-    # To grab the logs we only need a subset of the metadata from the server
-    obtain_and_print_logs(
-        config=config,
-        metadata_param=[
-            "id",
-            "executionStatus",
-            "backendLogs",
-            "subWorkflowMetadata",
-            "subWorkflowId",
-            "failures",
-        ],
-        status_params=status_param,
-        dont_expand_subworkflows=dont_expand_subworkflows,
-        print_logs=print_logs,
-    )
+    for workflow_id in workflow_ids:
+        command_setup_utils.resolve_workflow_id_and_server(
+            workflow_id=workflow_id, cromshell_config=config
+        )
 
-    return 0
+        task_logs = get_task_level_outputs(
+            config,
+            requested_status=status_param,
+            expand_subworkflows=not dont_expand_subworkflows,
+        )
 
+        if fetch_logs:
+            download_task_level_logs(all_task_log_metadata=task_logs)
+        else:
+            if json_summary:
+                io_utils.pretty_print_json(format_json=task_logs)
+            else:
+                print_task_level_logs(
+                    all_task_log_metadata=task_logs, cat_logs=print_logs
+                )
 
-def check_workflow_for_calls(workflow_status_json: dict) -> None:
-    """Check if the workflow has any calls"""
+    return return_code
 
-    if not workflow_status_json.get("calls"):
-        if workflow_status_json.get("failures"):
-            LOGGER.error(
-                "Empty 'calls' key found in workflow metadata. "
-                "Workflow failed with the following error(s): %s"
-                % workflow_status_json["failures"],
-            )
-            raise KeyError(
-                "Empty 'calls' key found in workflow metadata. "
-                "Workflow failed with the following error(s): %s"
-                % workflow_status_json["failures"],
-            )
-        else:
-            LOGGER.error(
-                "Empty 'calls' key found in workflow metadata. "
-                "This may indicate no tasks were run by the workflow, "
-                "the workflow has yet to run any tasks, or "
-                "a failure occurred before the workflow started."
-            )
-            raise KeyError(
-                "Empty 'calls' key found in workflow metadata."
-                "This may indicate no tasks were run by the workflow, "
-                "the workflow has yet to run any tasks, or "
-                "a failure occurred before the workflow started."
-            )
 
+def get_task_level_outputs(config, expand_subworkflows, requested_status) -> dict:
```

Comment:
Add reason why we're using Metadata api instead of Cromwells logs API as comment in code